### PR TITLE
Add a script to simplify getting relevant zap binary if not manually installed

### PR DESCRIPTION
**Pull request by andy31415**
We may have various automated tasks needing zap-cli:
   - CI of related projects (e.g. Pigweed CI)
   - setting up RaspberryPi environments (need ZAP_DEVELOPMENT_PATH set)
   - general developer convenience
      
Created a script that can be invoked as:

```
scripts/tools/zap/zap_download.py
# or
scripts/tools/zap/zap_download.py --zap source
```

and it downloads/sets up the given zap and also outputs the environment variable that needs to be set for the downloaded zap to be used. Not yet used in scripts, however expected longer term possible usage is that logging goes to stderr and shell scripts can eval stdout.

### CODE REVIEW

**Comment by arkq** [comment]
How about adding this to bootstrap as a check (e.g. install zap-cli if zap-cli is not available or if the version is not recent enough)? Today I've been struggling with updating my zap-cli because minimal version was bumped and I've forgotten how I've installed it on my system the last time... I forgot that the *.deb does not contain `zap-cli` binary...

**Review by bzbarsky-apple — APPROVED** [review]

File: scripts/tools/zap/zap_download.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    SOURCE = enum.auto()  # asking for a source download (will work on arm64 for example)
+
+
+def _GetDefaultExtractRoot():
+    if 'PW_ENVIRONMENT_ROOT' in os.environ:
+        return os.environ['PW_ENVIRONMENT_ROOT']
+    else:
+        # Before bootstrap, this will pick a temporary directory. Probably
+        # not ideal, but it likely just works
+        return '/tmp/'
```
[the remark this answers is not in the document]
Comment (andy31415) [reply scripts/tools/zap/zap_download.py:59]: I was unsure where to put this. I could put it inside the sdk checkout under .zap if that seems more useful or I could also raise an exception saying `you need to bootstrap`.

Do you have a preference?

[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply scripts/tools/zap/zap_download.py:59]: You don't necessarily need to bootstrap to get the env var, but you do need to activate, right?

I think falling back to sdk-checkout/.zap (and adding that to gitignore) would be fine.

[the remark this answers is not in the document]
Comment (andy31415) [reply scripts/tools/zap/zap_download.py:59]: right, activate or bootstrap.
I will update the default to .zap instead.

**Pull request merged**